zap/src/main/java/org/zaproxy/zap/view/MultipleRegexesOptionsPanel.java

```diff
@@ -76,6 +83,50 @@ public MultipleRegexesOptionsPanel(Dialog owner, RegexesTableModel model) {
 
         this.owner = owner;
         getTable().setSortOrder(0, SortOrder.ASCENDING);
+        JButton importButton =
+                new JButton(Constant.messages.getString("context.include.import.label"));
+        importButton.addActionListener(e -> importButtonClicked(e));
+        addButton(importButton);
+    }
+
+    public void importButtonClicked(ActionEvent event) {
+        File file = selectImportFile();
+        if (file != null) {
+            List<String> regexes = getRegexes();
+
+            String line = null;
+            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                while ((line = br.readLine()) != null) {
+                    String trimmedRegex = line.trim();
+                    if (!regexes.contains(trimmedRegex)) {
+                        Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
```

Comment:
This should be extracted to use when importing and adding.

Reply:
Extracted duplicate checking and valid pattern checking. Using extracted function for adding and importing of regexes.